scripts/run_driver_tests.psm1

```diff
@@ -67,178 +69,196 @@ function Invoke-Test
         $ArgumentsList += '-s'
     }
 
-    Write-Log "$TestName $ArgumentsList"
-    $Output = &$TestName $ArgumentsList
-    $TestName = $OriginalTestName
-    if ($LASTEXITCODE -ne 0) {
-        throw ("$TestName Failed with $LASTEXITCODE.")
-    }
-
-    # Parse output to check for errors.
-    if ($Output -eq $null) {
-        throw ("No output generated by $TestName.")
-    }
-    Out-String -InputObject $Output | Write-Log
-    $ParsedOutput = $Output.Split(" ")
-    if (-not (($null -ne $ParsedOutput) -and ($ParsedOutput.Length -ge 2))) {
-        throw ("Failed to parse output generated by $TestName.")
-    }
-    if ($ParsedOutput[$ParsedOutput.Length -2] -eq "failed") {
-        throw ("$TestName Test Failed.")
-    }
-    $ParsedOutput = $Output.Split([System.Environment]::NewLine)
-    foreach($line in $ParsedOutput) {
-        if ($line -clike "*FAILED:*") {
-            Write-Host $line
-            throw ("$TestName Test Failed.")
-        }
-    }
+    $JoinedArgumentsList = $ArgumentsList -join " "
+    $TestRunScript = ".\Run-Self-Hosted-Runner-Test.ps1"
+    & $TestRunScript `
+        -TestCommand $TestName `
+        -TestArguments $JoinedArgumentsList `
+        -TestHangTimeout $TestHangTimeout `
+        -UserModeDumpFolder $UserModeDumpFolder `
+        -NeedKernelDump $True `
+        -Verbose
 
     Write-Log "$TestName Passed" -ForegroundColor Green
```

Comment:
Make sure you've checked test failures are correctly detected, eg. and a failing assert in several places in xdp_test and connect_redirect_test

Reply:
Not sure I understand this comment.  Could you please clarify with an example (if possible)?

Reply:
Ah. I see what you mean.  So long as the tests themselves report a non-zero exit code in case of such failures, the failures are correctly detected.